test/functional/p2p_sendheaders.py

```diff
@@ -334,7 +335,7 @@ def test_nonnull_locators(self, test_node, inv_node):
                 if j == 0:
                     # Announce via inv
                     test_node.send_block_inv(tip)
-                    test_node.wait_for_getheaders()
+                    test_node.wait_for_getheaders(block_hash=expected_hash)
```

Comment:
oh this is no longer true after 05f7f31. we will only receive a getheaders the first time because of `m_inv_triggered_getheaders_before_sync`. found this issue when i ran the tests by clearing old getheaders as mentioned in https://github.com/bitcoin/bitcoin/pull/29736#discussion_r1545111761.

Reply:
Agreed, it looks like it was already the case that we were asserting the same `getheaders` message given it was never popped in this test.

I guess we can check it now as:

```python
if I is 0:
    test.node.wait_for_getheaders(block_hash=tip)
else:
    test_node.last_message.get("getheaders)" is None
```